subgrounds/client/sync.py

```diff
@@ -0,0 +1,452 @@
+          Subgraph: A generated class representing the subgraph and its entities
+        """
+
+        return self.load(url, save_schema, cache_dir, True)
+
+    def load_api(
+        self, url: str, save_schema: bool = False, cache_dir: str | None = None
+    ) -> Subgraph:
+        """Performs introspection on the provided GraphQL API ``url`` to get the
+         schema, stores the schema if ``save_schema`` is ``True`` and returns a
+         generated class representing the GraphQL endpoint with all its entities.
+
+        Args:
+          url: The url of the API.
+          save_schema: Flag indicating whether or not the schema should be saved
+           to disk.
+
+        Returns:
+          A generated class representing the subgraph and its entities
+        """
+
+        return self.load(url, save_schema, cache_dir, False)
+
+    def execute(
+        self,
+        req: DataRequest,
+        pagination_strategy: Type[PaginationStrategy] | None = LegacyStrategy,
+    ) -> DataResponse:
+        """Executes a :class:`DataRequest` and returns a :class:`DataResponse`.
+
+        Args:
+          req: The :class:`DataRequest` object to be executed.
+          pagination_strategy: A Class implementing the :class:`PaginationStrategy`
+            ``Protocol``. If ``None``, then automatic pagination is disabled.
+            Defaults to :class:`LegacyStrategy`.
+
+        Returns:
+          A :class:`DataResponse` object representing the response
+        """
+
+        try:
+            executor = self._execute(req, pagination_strategy)
+
+            doc = next(executor)
+            while True:
+                data = self._fetch(
+                    doc.url, {"query": doc.graphql, "variables": doc.variables}
+                )
+                doc = executor.send(DocumentResponse(url=doc.url, data=data))
+
+        except StopIteration as e:
+            return e.value
+
+    def execute_iter(
+        self,
+        req: DataRequest,
+        pagination_strategy: Type[PaginationStrategy] | None = LegacyStrategy,
+    ) -> Iterator[DocumentResponse]:
+        """Same as `execute`, except that an iterator is returned which will iterate
+        the data pages.
+
+        Args:
+          req: The :class:`DataRequest` object to be executed
+          pagination_strategy: A Class implementing the :class:`PaginationStrategy`
+           ``Protocol``. If ``None``, then automatic pagination is disabled.
+           Defaults to :class:`LegacyStrategy`.
+
+        Returns:
+          An iterator over the :class:`DocumentResponse` pages.
+
+        ⚠️ DOES NOT apply global transforms across multiple documents or their pages.
+         Since we yield each page as we get it, it's not possible to accurately perform
+         the transforms since we don't collect the pages. This means transforms
+         expecting multiple documents or pages of documents will be inaccurate.
+        """
+
+        with suppress(StopIteration):
```

Comment:
I'll add a comment for this but essentially, I don't need to do anything with the `StopIteration` since I yield the values as I get them. There's no value attached to the exception unlike normal `execute`.

TBH, `_iter` is quite different than normal `execute` and the design of this makes me irk a little bit